File: integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/NamedQueryMappedSuperClass.java
package io.quarkus.it.panache.reactive;

import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.NamedQuery;

import io.quarkus.hibernate.reactive.panache.PanacheEntity;

@MappedSuperclass
@NamedQuery(name = "NamedQueryMappedSuperClass.getAll", query = "from NamedQueryEntity")
public class NamedQueryMappedSuperClass extends PanacheEntity {
    public String superField;
}

File: integration-tests/hibernate-reactive-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/reactive/kotlin/NamedQueryMappedSuperClass.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheEntity
import jakarta.persistence.MappedSuperclass
import jakarta.persistence.NamedQuery

@MappedSuperclass
@NamedQuery(name = "NamedQueryMappedSuperClass.getAll", query = "from NamedQueryEntity")
open class NamedQueryMappedSuperClass : PanacheEntity() {
    lateinit var superField: String
}
